File: 2037-minimum-number-of-moves-to-seat-everyone/counting-sort/counting-sort.py
from typing import List

class Solution:
  def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
    m = max(seats + students)
    res = 0
    count = [0] * (m + 1)
    for seat in seats:
      count[seat] += 1
    for student in students:
      count[student] -= 1
    for i in range(1, m + 1):
      count[i] += count[i - 1]
      res += abs(count[i])
    return res

File: 2037-minimum-number-of-moves-to-seat-everyone/counting-sort/counting-sort.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
  int minMovesToSeat(vector<int> &seats, vector<int> &students) {
    int m = max(*max_element(seats.begin(), seats.end()),
                *max_element(students.begin(), students.end())),
        res = 0;
    vector<int> count(m + 1, 0);
    for (int seat : seats)
      count[seat]++;
    for (int student : students)
      count[student]--;
    for (int i = 1; i <= m; i++) {
      count[i] += count[i - 1];
      res += abs(count[i]);
    }
    return res;
  }
};
